lib/ansible/modules/cloud/vmware/vmware_guest_network.py

```diff
@@ -461,9 +479,31 @@ def get_network_config_spec(self, vm_obj, network_list):
                             if 'connected' in network and nic_device.connectable.connected != network['connected']:
                                 nic_device.connectable.connected = network['connected']
                                 self.change_detected = True
-                            if 'name' in network and nic_device.deviceInfo.summary != network['name']:
-                                nic_device.deviceInfo.summary = network['name']
-                                self.change_detected = True
+                            if 'name' in network:
+                                network_object = self.find_network_by_name(network_name=network['name'])[0]
+                                if network_object and hasattr(network_object, 'portKeys'):
```

Comment:
When changing portgroup to dvportgroup under this condition error occurs.
The cause is that the `port` key does not exist when the portgroup is assigned to the vNIC.
The following conditions are necessary.

```suggestion
                                if network_object and hasattr(network_object, 'portKeys') and hasattr(nic_spec.device.backing, 'port'):
```